### PR TITLE
Make sure TcpClient.ExclusiveAddressUse throws on Unix

### PR DESCRIPTION
**Pull request by svick**
This is related to the failure that happened in #9849.

cc: @stephentoub @ericeil

### CODE REVIEW

File: src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.Unix.cs [hunk 1]
```diff
@@ -378,10 +378,9 @@ private bool ExclusiveAddressUseCore
                 ShadowOptions so = EnsureShadowValuesInitialized();
                 so._exclusiveAddressUse = value ? 1 : 0;
                 so._exclusiveAddressUseInitialized = true;
-                if (_clientSocket != null)
-                {
-                    _clientSocket.ExclusiveAddressUse = value; // Use setter explicitly as it does additional validation beyond that done by SetOption
-                }
+
+                Socket s = _clientSocket ?? CreateSocket();
```
Comment (stephentoub) [inline src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.Unix.cs]: If _clientSocket is null, we should be disposing of the temporary socket that gets created.

**Comment by stephentoub** [comment]
One minor issue, otherwise LGTM.  Thanks for fixing this.

Subsequently, we might want to consider making get_ExclusiveAddressUse (both on TcpClient and on Socket) return a hardcoded Boolean value rather than throwing.

File: src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.Unix.cs (hunk 1 above)
Comment (svick) [inline src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.Unix.cs]: Fixed.

**Comment by stephentoub** [comment]
LGTM. Thanks.

**Pull request merged**